src/catalogue/category/catalogueCategoryDialog.component.tsx

```diff
@@ -12,636 +16,861 @@ import {
   FormHelperText,
   FormLabel,
   Grid,
+  IconButton,
   Radio,
   RadioGroup,
+  Stack,
   TextField,
+  Tooltip,
   Typography,
 } from '@mui/material';
 import { AxiosError } from 'axios';
 import React from 'react';
-import { APIError } from '../../api/api.types';
 import {
-  useAddCatalogueCategory,
-  useEditCatalogueCategory,
+  Controller,
+  FormProvider,
+  useFieldArray,
+  useForm,
+  useFormContext,
+} from 'react-hook-form';
+import {
+  AllowedValues,
+  AllowedValuesListType,
+  APIError,
+  CatalogueCategory,
+  CatalogueCategoryPost,
+  CatalogueCategoryPostProperty,
+  CatalogueCategoryProperty,
+  CatalogueCategoryPropertyType,
+} from '../../api/api.types';
+import {
+  usePatchCatalogueCategory,
+  usePostCatalogueCategory,
 } from '../../api/catalogueCategories';
+import { useGetUnits } from '../../api/units';
 import {
-  AddCatalogueCategory,
-  AddCatalogueCategoryProperty,
   AddCatalogueCategoryPropertyWithPlacementIds,
   AddCatalogueCategoryWithPlacementIds,
-  AllowedValuesListErrorsType,
-  CatalogueCategory,
-  CatalogueItemPropertiesErrorsType,
-  EditCatalogueCategory,
+  AllowedValues as AllowedValuesPlaceholder,
 } from '../../app.types';
+import { CatalogueCategorySchema, RequestType } from '../../form.schemas';
 import handleIMS_APIError from '../../handleIMS_APIError';
-import { generateUniqueId, trimStringValues } from '../../utils';
-import CataloguePropertiesForm from './cataloguePropertiesForm.component';
+import CatalogueItemsPropertiesTable from './property/catalogueItemPropertiesTable.component';
 
 // Function to convert a list of strings to a list of numbers
 export const convertListToNumbers = (values: string[]): number[] => {
   return values.map((value) => parseFloat(value));
 };
+
 export interface CatalogueCategoryDialogProps {
   open: boolean;
   onClose: () => void;
   parentId: string | null;
-  type: 'add' | 'edit' | 'duplicate';
+  requestType: RequestType;
+  duplicate?: boolean;
   selectedCatalogueCategory?: CatalogueCategory;
   resetSelectedCatalogueCategory: () => void;
 }
```

Comment:
Could this be moved below the transform functions so its right above the component.

Reply:
I don't understand what you want me to d. I though i did it